plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -1085,8 +1102,18 @@ export const coreProfilerStateMachineDefinition = createMachine( {
 						onDone: [
 							{
 								target: 'pluginsSkipped',
-								cond: ( context, event ) =>
-									event.data.length === 0,
+								cond: ( context, event ) => {
+									// Skip the plugins page
+									// When there is 0 plugin returned from the server
+									// Or all the plugins are activated already.
+									return (
+										event.data.length === 0 ||
+										event.data.filter(
+											( plugin: Extension ) =>
+												plugin.is_activated === false
+										).length === 0
```

Comment:
Nit: I think we can just replace `filter` with the `every` method, and we don't need to check `length`.

```js
event.data.every( ( plugin ) => plugin.is_activated )
```

Reply:
nice, that works!